LEGO1/mxstring.cpp

```diff
@@ -79,3 +79,37 @@ const MxString &MxString::operator=(const char *param)
 
   return *this;
 }
+
+// Return type is intentionally just MxString, not MxString&.
+// This forces MSVC to add $ReturnUdt$ to the stack for 100% match.
+// OFFSET: LEGO1 0x100ae580
+MxString MxString::operator+(const char *str)
+{
+  // MxString constructor calls malloc(1) for m_data, so free that first
+  MxString tmp;
+  free(tmp.m_data);
+
+  tmp.m_length = strlen(str) + this->m_length;
+  tmp.m_data = (char *)malloc(tmp.m_length + 1);
```

Comment:
I think all allocations and deallocations of `m_data` in `MxString` should use `operator new[]` and `operator delete[]` because this is calling the same allocation function when creating objects.

Reply:
Just changed that and it fixed the swapped registers in the other operator function